### PR TITLE
new stream functionality to read in ndep if it is not calculated by WACCM

### LINKED ISSUE #104: Always send Nitrogen-deposition to surface components
In standard configurations both CTSM and the ocean model need to have Nitrogen-deposition used. The system is currently configured so that CTSM and POP assume that they have to provide ndep, but in some cases CAM sends ndep down to them (when WACCM is being used) -- and when it does they use it rather than providing it themselves. However, at the script level they don't know that it's being sent down so normally will abort if they don't have a file to read ndep setup. In the case when CAM IS sending ndep to the surface components they don't need to abort, and we'd like a way for them to know they don't need to abort when setting up their namelists. The other bad thing about the current situation is that the surface components will set control variables for the ndep file -- that won't actually be used. In this setup they are dummy variables that a user might think matters -- but actually doesn't. So it's bad behavior (to abort) and it's bad documentation of the case. The surface components could query the compset -- but since whether ndep is sent is a run-time configurable variable for CAM -- that's NOT a good solution.

The long term best answer to this to ALWAYS send ndep down to the surface components from CAM. When WACCM is on it'll be prognostic, but for other configuration's it'll be prescribed from datasets coming from previous simulations.

Note this relates to: #102

CTSM issue that relates to this: https://github.com/ESCOMP/CTSM/issues/946

### PR DESCRIPTION
**Pull request by mvertens**
This PR addresses issue #104 and enables CAM to read in ndep (using the CDEPS inline API) from forcing files if it is not computing NDEP prognostically
As a result - CAM will ALWAYS send NDEP to the mediator
Right how, CLM is not using the passed NDEP from the streams - since it is still reading the drv_flds_in for NDEP
Once this PR is accepted the drv_flds_in entry for ndep can be removed and CLM can always accept NDEP from either CAM or DATM.
New XML variables are introduced to specify the stream forcing. This is being done in CTSM as well for all of its CDEPS stream specific variables.

Closes #104

### CODE REVIEW

**Comment by mvertens** [comment]
@gold2718 @fvitt - yes it is #104. Sorry I did not point to that.

**Comment by mvertens** [comment]
@cacraigucar @nusbaume @gold2718 - just a clarification that this PR is needed in order for CAM to always send NDEP to CLM. As @adamrher has already encountered this is currently a requirement in spin up runs that datm expects NDEP values in the coupler history files. A separate PR where CLM no longer will need to read in NDEP from its own stream files but rather listen to the NDEP from CAM and DATM can occur once this PR is accepted.

**Review by cacraigucar — COMMENTED** [review]
My first questions  - I've not completed my review, but getting some answers to these questions might help me with the rest of the review

File: bld/build-namelist [hunk 1]
```diff
@@ -4023,11 +4040,37 @@ add_default($nl, 'cam_snapshot_before_num');
...
+  }
+}
+
+
 #-----------------------------------------------------------------------------------------------
 # Write output files
 
 # Get array of group names that have been defined in the namelist.
 my @nl_groups = $nl->get_group_names();
+print "DEBUG: nl_groups are @nl_groups \n";
```
[the remark this answers is not in the document]
Comment (mvertens) [reply bld/build-namelist:4073]: @cacraig - thanks for looking at this PR. No - this was an error on my part and should be removed.

**Comment by mvertens** [comment]
@cacraigucar - I used the wrong branch for this PR. I will close it and reopen a PR with the correct branch.

**Comment by mvertens** [comment]
I used the wrong branch for this PR. I will close it and reopen a PR with the correct branch.

**Pull request closed without merging**